src/components/common/ProductBox/ProductBox.js

```diff
@@ -11,64 +11,94 @@ import {
 import { faStar as farStar, faHeart } from '@fortawesome/free-regular-svg-icons';
 import Button from '../Button/Button';
 
-const ProductBox = ({ name, image, price, oldPrice, promo, stars }) => (
-  <div className={styles.root}>
-    <div className={styles.photo}>
-      <img src={image} alt={name} />
-      {promo && <div className={styles.sale}>{promo}</div>}
-      <div className={styles.buttons}>
-        <Button variant='small'>Quick View</Button>
-        <Button variant='small'>
-          <FontAwesomeIcon icon={faShoppingBasket}></FontAwesomeIcon> ADD TO CART
-        </Button>
-      </div>
-    </div>
-    <div className={styles.content}>
-      <h5>{name}</h5>
-      <div className={styles.stars}>
-        {[1, 2, 3, 4, 5].map(i => (
-          <a key={i} href='#'>
-            {i <= stars ? (
-              <FontAwesomeIcon icon={faStar}>{i} stars</FontAwesomeIcon>
-            ) : (
-              <FontAwesomeIcon icon={farStar}>{i} stars</FontAwesomeIcon>
-            )}
-          </a>
-        ))}
-      </div>
-    </div>
-    <div className={styles.line}></div>
-    <div className={styles.actions}>
-      <div className={styles.outlines}>
-        <Button variant='outline'>
-          <FontAwesomeIcon icon={faHeart}>Favorite</FontAwesomeIcon>
-        </Button>
-        <Button variant='outline'>
-          <FontAwesomeIcon icon={faExchangeAlt}>Add to compare</FontAwesomeIcon>
-        </Button>
+const ProductBox = ({
+  id,
+  name,
+  image,
+  price,
+  oldPrice,
+  promo,
+  stars,
+  setCompare,
+  count,
+  compare,
+}) => {
+  const compareHandler = event => {
+    event.preventDefault();
+    const isExist = compare.products.reduce(
+      (accumulator, product) => accumulator || product.id === id,
+      false
+    );
+    if (isExist !== true)
+      // prevent to add the same product to compare
+      count < 4
+        ? setCompare({ id, image })
+        : alert('You can compare maximum of 4 products!');
```

Comment:
Zapisz tą 4 pod stałą i użyj jej w kodzie i w komunikacie wyświetlanym dla użytkownika. łatwiej to ędize zmienić w przyszłości